common/tools/dev-tool/src/commands/run/update-snippets.ts

```diff
@@ -0,0 +1,530 @@
+import fs from "node:fs/promises";
+import { existsSync } from "node:fs";
+import { EOL } from "node:os";
+import path from "node:path";
+import ts from "typescript";
+import { leafCommand, makeCommandInfo } from "../../framework/command";
+import { findMatchingFiles } from "../../util/findMatchingFiles";
+import { format } from "../../util/prettier";
+import { createPrinter } from "../../util/printer";
+import { ProjectInfo, resolveProject } from "../../util/resolveProject";
+import { convert } from "../../util/samples/convert";
+import { testSyntax } from "../../util/samples/syntax";
+import { createDiagnosticEmitter } from "../../util/typescript/diagnostic";
+
+export const commandInfo = makeCommandInfo(
+  "update-snippets",
+  "find README and TSDoc snippets throughout the package and update their contents.",
+  {},
+);
+
+const log = createPrinter("update-snippets");
+
+const SNIPPET_PATH = ["test", "snippets.spec.ts"];
+
+/**
+ * Describes a location where a snippet is actually presented to a reader, i.e. a Markdown file or JSDoc comment
+ * location.
+ */
+interface SnippetLocationInfo {
+  /** The snippet name determined by the tag on its enclosing fence. */
+  name: string;
+  /** The fence string that started this snippet. */
+  fence: string;
+  /** The target language of the snippet. */
+  language: "js" | "ts";
+  /** The absolute path to the file the snippet is located in. */
+  absoluteFilePath: string;
+  /** The contents of the snippet with the line prefix stripped */
+  lineTrimmedContents: string[];
+  /** The range of lines in the file where the snippet appears. */
+  lineRange: [number, number];
+  /** The prefix that appeared before the code fence, which will be inserted before each line in the snippet. */
+  linePrefix: string;
+}
+
+/**
+ * Finds all candidate files that _might_ contain snippet locations.
+ *
+ * @param dir - the base directory of the package
+ */
+async function* getAllSnippetFiles(dir: string): AsyncIterable<string> {
+  // Only consider markdown files up to a depth of 1 (i.e. _in_ the project folder). This is to prevent grabbing things
+  // like samples/*/README.md and other similar files that are not really part of the "source" of the package.
+  yield* findMatchingFiles(dir, (name) => name.endsWith(".md"), { maxDepth: 1 });
```

Comment:
I almost wonder if to start we can just focus on `src/README.md` - are there other files you are thinking will need to be updated?

Reply:
I'd rather focus on all markdown files, for example in identity, there is a troubleshooting guide which had some issues as well as other markdown files.  I'd rather have them scan all files, and we can opt out using the ignore.